consensus/src/sync/live/block_queue/queue.rs

```diff
@@ -865,6 +869,8 @@ impl<N: Network> Stream for BlockQueue<N> {
     type Item = QueuedBlock<N>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
+        self.waker.store_waker(cx);
```

Comment:
We only have to store the waker in the very bottom of this function, if we return `Poll::Pending`.